probes/testsRunInCI/impl.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint:stylecheck
+package testsRunInCI
+
+import (
+	"embed"
+	"fmt"
+	"strings"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/finding"
+	"github.com/ossf/scorecard/v4/probes/internal/utils/uerror"
+)
+
+//go:embed *.yml
+var fs embed.FS
+
+const (
+	Probe   = "testsRunInCI"
+	success = "success"
+)
+
+func Run(raw *checker.RawResults) ([]finding.Finding, string, error) {
+	if raw == nil {
+		return nil, "", fmt.Errorf("%w: raw", uerror.ErrNil)
+	}
+
+	var findings []finding.Finding
+
+	c := raw.CITestResults
+
+	if len(c.CIInfo) == 0 {
+		f, err := finding.NewWith(fs, Probe,
+			"no pull requests found", nil,
+			finding.OutcomeNotApplicable)
+		if err != nil {
+			return nil, Probe, fmt.Errorf("create finding: %w", err)
+		}
+		findings = append(findings, *f)
+		return findings, Probe, nil
+	}
+
+	totalMerged := 0
+	totalTested := 0
+	for i := range c.CIInfo {
+		r := c.CIInfo[i]
+		totalMerged++
+
+		// GitHub Statuses.
+		prSuccessStatus, f, err := prHasSuccessStatus(r)
+		if err != nil {
+			return nil, Probe, fmt.Errorf("create finding: %w", err)
+		}
+		if prSuccessStatus {
+			findings = append(findings, *f)
+			totalTested++
+			continue
+		}
+
+		// GitHub Check Runs.
+		prCheckSuccessful, f, err := prHasSuccessfulCheck(r)
+		if err != nil {
+			return nil, Probe, fmt.Errorf("create finding: %w", err)
+		}
+		if prCheckSuccessful {
+			findings = append(findings, *f)
+			totalTested++
+		}
+	}
+
+	for i := range findings {
+		f := &findings[i]
+		f.WithValues(map[string]int{
+			"totalMerged": totalMerged,
+			"totalTested": totalTested,
+		})
+	}
```

Comment:
I was under the impression the Values were going to replace having `n` findings, leading to 1 finding whose values describes the stats.

also, lets make the keys exported consts as well, so that callers dont need to redefine the key

Reply:
AFAIR, the reason here (and in a few other probes) for creating multiple findings is to enable logging of multiple pieces of metadata.

Reply:
What additional metadata is provided for this case?

Reply:
file location:
```go
		loc := &finding.Location{
			Path: status.URL,
			Type: finding.FileTypeURL,
		}
		f = f.WithLocation(loc)
```
https://github.com/ossf/scorecard/pull/3621/files#diff-3abb84416a867919c8dc78c8e132c33f3a96c78a4516252d9f07afae10f87c05R114-R118

and

https://github.com/ossf/scorecard/pull/3621/files#diff-3abb84416a867919c8dc78c8e132c33f3a96c78a4516252d9f07afae10f87c05R147-R151

Reply:
reversible decision, so fine with either and we can change it later if needed

Reply:
Closing since current state of the PR is fine.

Reply:
these values need to be removed though.

Reply:
Done in https://github.com/ossf/scorecard/pull/3621/commits/a7970c6231277f2d808da2716c3e28f8a85b6948